src/shareddoc.js

```diff
@@ -430,26 +427,17 @@ export class WSSharedDoc extends Y.Doc {
  * @param {boolean} gc - whether garbage collection is enabled
  * @returns The Yjs document object, which may be shared across multiple sockets.
  */
-export const getYDoc = async (docname, conn, env, storage, timingData, gc = true) => {
-  let doc = docs.get(docname);
-  if (doc === undefined) {
-    // The doc is not yet in the cache, create a new one.
-    doc = new WSSharedDoc(docname);
-    doc.gc = gc;
-    docs.set(docname, doc);
-  }
+export const createYDoc = async (docname, conn, env, storage, timingData, docsCache, gc = true) => {
+  const doc = new WSSharedDoc(docname);
+  doc.gc = gc;
 
   if (!doc.conns.get(conn)) {
     doc.conns.set(conn, new Set());
   }
 
   // Store the service binding to da-admin which we receive through the environment in the doc
   doc.daadmin = env.daadmin;
-  if (!doc.promise) {
-    // The doc is not yet bound to the persistence layer, do so now. The promise will be resolved
-    // when bound.
-    doc.promise = persistence.bindState(docname, doc, conn, storage);
-  }
+  doc.promise = persistence.bindState(docname, doc, conn, storage, docsCache);
```

Comment:
I think this is not doing what happened before. IIRC, the idea was to use the promise to make sure the doc doesn't get bound to persistence called concurrently. @bosschaert, are you sure this is still correct (I would have expected to have this move up to the caller like it used to for the await)...